File: deps/src/examples/hello.cpp
#include <string>

#include <cxx_wrap.hpp>

std::string greet()
{
   return "hello, world";
}

JULIA_CPP_MODULE_BEGIN(registry)
  cxx_wrap::Module& hello = registry.create_module("CppHello");
  hello.method("greet", &greet);
JULIA_CPP_MODULE_END

File: test/hello.jl
# Wrap the functions defined in C++
module CppHello
  include(joinpath(@__DIR__, "testcommon.jl"))
  @wrapmodule CxxWrap.CxxWrapCore.libhello()

  function __init__()
    @initcxx
  end
end

using Test

# Output:
@show CppHello.greet()

@testset "$(basename(@__FILE__)[1:end-3])" begin

# Test the result
@test CppHello.greet() == "hello, world"

end